### PR TITLE
[Impeller Scene] Node deserialization

### PR DESCRIPTION
**Pull request by bdero**
Utils to run flatbuffer validation and unpack nodes from ipscenes.

### CODE REVIEW

File: impeller/scene/mesh.h [hunk 1]
```diff
@@ -18,19 +18,20 @@ namespace scene {
...
   ~Mesh();
 
   void AddPrimitive(Primitive mesh_);
+  std::vector<Primitive>& GetPrimitives();
 
   bool Render(SceneEncoder& encoder, const Matrix& transform) const;
 
  private:
-  std::vector<Primitive> meshes_;
+  std::vector<Primitive> primitives_;
```
Comment (chinmaygarde) [inline impeller/scene/mesh.h:34]: Nit: `DISALLOW_COPY_AND_ASSIGN`

File: impeller/scene/mesh.h [hunk 2]
```diff
@@ -18,19 +18,20 @@ namespace scene {
...
-    std::shared_ptr<Material> material_;
+    std::shared_ptr<Geometry> geometry;
+    std::shared_ptr<Material> material;
   };
 
   Mesh();
   ~Mesh();
 
   void AddPrimitive(Primitive mesh_);
+  std::vector<Primitive>& GetPrimitives();
```
Comment (chinmaygarde) [inline impeller/scene/mesh.h:29]: Nit: `const`

**Review by chinmaygarde — APPROVED** [review]

File: impeller/scene/mesh.h (hunk 1 above)
Comment (bdero) [reply impeller/scene/mesh.h:37]: Done.

File: impeller/scene/mesh.h (hunk 2 above)
Comment (bdero) [reply impeller/scene/mesh.h:32]: This one's intentionally mutable.

**Comment by bdero** [comment]
Is the pipeline flake rate seems to have gotten dramatically higher over the past day.

**Comment by zanderso** [comment]
@chinmaygarde can we land the workaround on Monday?

**Comment by chinmaygarde** [comment]
Yeah. Will do.

**Pull request merged**